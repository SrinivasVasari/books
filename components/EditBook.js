import React, { useState, useContext } from "react";
import BooksContext from "../context/context";

const EditBook = () => {
  const { state, dispatch } = useContext(BooksContext);
  const [bookName, setBookName] = useState(state.currentBook.books.bookName);
  const [bookAuthor, setBookAuthor] = useState(
    state.currentBook.books.bookAuthor
  );
  const [bookDesc, setBookDesc] = useState(state.currentBook.books.bookDesc);

  const handleSubmit = (event) => {
    event.preventDefault();
    const updateBook = {
      bookName,
      bookAuthor,
      bookDesc,
    };
    if (bookName.trim() === "") {
      alert("Cannot add blank book");
    } else {
      dispatch({ type: "UPDATE_BOOK", payload: updateBook });
      setBookName("");
      setBookAuthor("");
      setBookDesc("");
    }
  };

  return (
    <>
      <h4>Update Book</h4>
      <div className="editBookForm row">
        <form className="col s12" onSubmit={handleSubmit}>
          <div class="row">
            <div class="input-field col s6">
              <input
                type="text"
                name="Name"
                id="Name"
                placeholder="Book Name"
                onChange={(e) => setBookName(e.target.value)}
                value={bookName}
                className="validate"
              />
            </div>
            <div class="input-field col s6">
              <input
                type="text"
                name="Author"
                id="Author"
                placeholder="Book Author"
                onChange={(e) => setBookAuthor(e.target.value)}
                value={bookAuthor}
                className="validate"
              />
            </div>
          </div>
          <div class="row">
            <div class="input-field col s12">
              <input
                type="text"
                onChange={(e) => setBookDesc(e.target.value)}
                value={bookDesc}
                name="Description"
                id="Description"
                placeholder="Book Description"
                className="validate"
              />
            </div>
          </div>
          <button type="submit" className="waves-effect waves-light btn-small">
            update Book
          </button>
        </form>
      </div>
    </>
  );
};

export default EditBook;
