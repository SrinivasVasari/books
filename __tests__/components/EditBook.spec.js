import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import EditBook from "../../components/EditBook";
import BooksContext from "../../context/context";

describe("Add Book Container", () => {
  const testState = {
    bookName: "test",
    bookAuthor: "test1",
    bookDesc: "test2",
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EditBook />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should call input onChange", () => {
    wrapper
      .find("input")
      .at(0)
      .simulate("change", { target: { name: "name", value: "test" } });
    wrapper
      .find("input")
      .at(1)
      .simulate("change", { target: { name: "author", value: "test1" } });
    wrapper
      .find("input")
      .at(2)
      .simulate("change", {
        target: { name: "description", value: "test2" },
      });
    expect(testState.bookName).toEqual("test");
    expect(testState.bookAuthor).toEqual("test1");
    expect(testState.bookDesc).toEqual("test2");
  });
  it("calls handle Submit function when form is submitted", () => {
    const dispatchMock = jest.fn();
    const handleSubmitFn = jest.fn();
    const fakeEvent = { preventDefault: () => {} };
    const wrapper = mount(
      <BooksContext.Provider value={{ dispatch: dispatchMock }}>
        <EditBook onSubmit={handleSubmitFn} />
      </BooksContext.Provider>
    );
    const form = wrapper.find("form");
    wrapper.find("#name").simulate("change", {
      target: { name: "name", value: "blah" },
    });
    wrapper.find("#author").simulate("change", {
      target: { name: "author", value: "blah1" },
    });
    wrapper.find("#description").simulate("change", {
      target: { name: "description", value: "blah2" },
    });
    form.simulate("submit", fakeEvent);
    expect(handleSubmitFn.mock.calls).not.toBeNull();
  });
});
