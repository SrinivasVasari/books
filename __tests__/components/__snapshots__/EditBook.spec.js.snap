// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Book Container should render correctly 1`] = `
<form
  className="form col s12"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="input-field col s6"
    >
      <input
        className="name"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Book Name"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-field col s6"
    >
      <input
        className="author"
        id="author"
        name="author"
        onChange={[Function]}
        placeholder="Book Author"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <input
        className="description"
        id="description"
        name="description"
        onChange={[Function]}
        placeholder="Book Description"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="waves-effect waves-light btn-small"
    type="submit"
  >
    update Book
  </button>
</form>
`;
